### PR TITLE
Basic community fixes PR

### PR DESCRIPTION
**Pull request by th3w1zard1**
This PR will continually be updated with the smaller community issues I've seen while cleaning up the older issues.
(before anyone gets too excited - commit `e6b3e6e` was reverted as it wouldn't work with non-ha switches)

- #423 
- #378 
- #403
- #461
- #408 
- #80

### CODE REVIEW

File: custom_components/adaptive_lighting/switch.py [hunk 1]
```diff
@@ -553,7 +553,8 @@ def _expand_light_groups(hass: HomeAssistant, lights: list[str]) -> list[str]:
 
 def _supported_features(hass: HomeAssistant, light: str):
     state = hass.states.get(light)
-    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    # see #423, I guess not all lights have ATTR_SUPPORTED_FEATURES
+    supported_features = state.attributes.get(ATTR_SUPPORTED_FEATURES, {})
```
Comment (basnijholt) [inline custom_components/adaptive_lighting/switch.py:557]: I think this is incorrect. `state.attributes[ATTR_SUPPORTED_FEATURES]` is an `enum.IntFlag` which is a subtype of `int`. So I think the correct thing is to do `state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)`

File: custom_components/adaptive_lighting/switch.py [hunk 2]
```diff
@@ -952,7 +953,8 @@ async def _adapt_light(
         if prefer_rgb_color is None:
             prefer_rgb_color = self._prefer_rgb_color
 
-        if "transition" in features:
+        # Check transition == 0 to fix #378
+        if "transition" in features and transition != 0:
```
Comment (basnijholt) [inline custom_components/adaptive_lighting/switch.py:957]: `        if "transition" in features and transition > 0:`

File: custom_components/adaptive_lighting/switch.py (hunk 1 above)
Comment (th3w1zard1) [reply custom_components/adaptive_lighting/switch.py:557]: Please take a look at the issue commented above that change.

Comment (basnijholt) [reply custom_components/adaptive_lighting/switch.py:557]: I see what that person reported, but in the case that `ATTR_SUPPORTED_FEATURES` is indeed missing. The following happens:
```python
supported_features = {}
value = 1
supported_features & value
```
which will raise:
```
>>> supported_features & value
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for &: 'dict' and 'int'
```

Comment (th3w1zard1) [reply custom_components/adaptive_lighting/switch.py:557]: Hmm... interesting. [I just compared to the older releases](https://github.com/basnijholt/adaptive-lighting/compare/1.3.0...1.6.0) that released around the time that he posted, and confirmed no code changed there...

Your check is great! I'm just wondering what happened to that person.

**Pull request merged**